usr/src/uts/common/fs/zfs/dsl_dataset.c

```diff
@@ -2151,6 +2185,14 @@ dsl_dataset_rollback_check(void *arg, dmu_tx_t *tx)
 		return (SET_ERROR(EINVAL));
 	}
 
+	/* XXX No rollback to a snapshot created in the current txg. */
```

Comment:
Okay, I will.  The reason is essentially the same: we do not to create any dirty blocks in the same txg as the snapshot was created _after_ the snapshot was created.